src/coreclr/vm/eetwain.cpp

```diff
@@ -2277,13 +2281,78 @@ ULONG32 EECodeManager::GetStackParameterSize(EECodeInfo * pCodeInfo)
 
 #ifdef FEATURE_INTERPRETER
 
-bool InterpreterCodeManager::UnwindStackFrame(PREGDISPLAY     pContext,
+static void VirtualUnwindInterpreterCallFrame(TADDR sp, T_CONTEXT *pContext)
+{
+    PTR_InterpMethodContextFrame pFrame = dac_cast<PTR_InterpMethodContextFrame>(sp);
+    pFrame = pFrame->pParent;
+    if (pFrame != NULL)
+    {
+        SetIP(pContext, (TADDR)pFrame->ip);
+        SetSP(pContext, dac_cast<TADDR>(pFrame));
+    }
+    else
+    {
+        // This indicates that there are no more interpreter frames to unwind in the current InterpExecMethod
+        // The stack walker will not find any code manager for the address 0 and move on to the next explicit
+        // frame which is the InterpreterFrame.
+        // Interpreter-TODO: Consider returning the context of the JITted / AOTed code that called the interpreter instead
+        SetIP(pContext, 0);
+        SetSP(pContext, sp);
+    }
+    pContext->ContextFlags = CONTEXT_CONTROL;
+}
+
+bool InterpreterCodeManager::UnwindStackFrame(PREGDISPLAY     pRD,
                                               EECodeInfo     *pCodeInfo,
                                               unsigned        flags,
                                               CodeManState   *pState)
 {
-    // Interpreter-TODO: Implement this
-    return false;
+    if (pRD->IsCallerContextValid)
+    {
+        // We already have the caller's frame context
+        // We just switch the pointers
+        PT_CONTEXT temp      = pRD->pCurrentContext;
```

Comment:
Isn't unwinding the process of walking up the stack? Why are we switching the pointers instead of just overwriting the current context with the caller context? I assume I'm missing something here

Reply:
If we already have the caller context available, then you can just set the current context to it, there is no need to call unwind again to get the same values. The switching of pointer is to avoid copying in this specific case. 
There are cases when we want to know the caller context without actually unwinding, so we call EnsureCallerContextIsValid. That's where the caller context gets created and then it can get reused here.